src/map.cpp

```diff
@@ -1097,14 +1100,9 @@ bool Map::isBlockOccluded(MapBlock *block, v3s16 cam_pos_nodes)
 	// sqrt(1^2 + 1^2 + 1^2) = 1.732
 	float end_offset = -BS * MAP_BLOCKSIZE * 1.732f;
 
-	// to reduce the likelihood of falsely occluded blocks
-	// require at least two solid blocks
-	// this is a HACK, we should think of a more precise algorithm
-	u32 needed_count = 2;
-
 	for (const v3s16 &dir : dir9) {
 		if (!isOccluded(cam_pos_nodes, pos_blockcenter + dir, step, stepfac,
-				start_offset, end_offset, needed_count))
+				start_offset, end_offset, 1))
```

Comment:
If the new technique doesn't require `needed_count`, this extra param could just be removed altogether?

Reply:
Yep. I did not want to confuse the change too much.

Reply:
Sorry I meant I want to keep the change as small as possible. If this works out I'll remove the count nonsense completely.

Reply:
Yay :)